src/Pages/GameScreen/GameStepper.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState, useEffect, MouseEvent } from "react";
+import { navigate } from "@reach/router";
+import { RoutPaths } from "Pages/routes";
+import arrayShuffle from "array-shuffle";
+import { CurrentScoreType, CurrentVariants, GameStepperProps } from "types";
+import { DEFAULT_CURRENT_SCORE } from "Pages/EndScreens/GameOver";
+import { getStepCounter, getUpdatedUserFlags } from "gameHandlers";
+import {
+  getDataFromLocalStorage,
+  StorageKeys,
+} from "services/getDataFromLocalStorage";
+
+export const GameStepper = ({
+  userFlags,
+  setUserFlags,
+  correctVariant,
+  answerVariants,
+  allCountryFlagsLength,
+}: GameStepperProps) => {
+  const [score, setScore] = useState<CurrentScoreType>(() => {
+    return getDataFromLocalStorage<CurrentScoreType>(
+      StorageKeys.CurrentScore,
+      DEFAULT_CURRENT_SCORE
+    );
+  });
+  const [stepCounter, setStepCounter] = useState(() => getStepCounter());
+
+  const handleChosenAnswer = (event: MouseEvent<HTMLButtonElement>) => {
+    // @ts-ignore
+    const chosenAnswer = event.target.innerText;
+
+    if (chosenAnswer === correctVariant.name.toUpperCase()) {
+      const updatedUserFlags = getUpdatedUserFlags(userFlags, correctVariant);
+
+      setUserFlags(updatedUserFlags);
+      setScore({ time: new Date().toString(), score: score.score + 10 });
+      setStepCounter(stepCounter + 1);
+    } else {
+      navigate(RoutPaths.GameOver);
+    }
+  };
+
+  useEffect(() => {
+    if (userFlags.length === 0) {
+      navigate(RoutPaths.WinScreen);
+    }
+  }, [userFlags.length]);
+
+  useEffect(() => {
+    const userCurrentScore: CurrentScoreType = {
+      time: new Date().toString(),
+      score: score.score,
+    };
+
+    localStorage.setItem(
+      StorageKeys.CurrentScore,
+      JSON.stringify(userCurrentScore)
+    );
+    localStorage.setItem(StorageKeys.StepCounter, JSON.stringify(stepCounter));
+  }, [score, stepCounter]);
+
+  useEffect(() => {
+    const currentVariants: CurrentVariants = {
+      correct: correctVariant.id,
+      answerVariants: answerVariants.map((item) => item.id),
```

Comment:
just not to be confused, it is better to use different names here, like answerVariantsIds

Reply:
fixed 
variables names were changed to make code more clear